### PR TITLE
Unit test speed optimization

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #DNE
# What's Changed

- While testing fixes for #3930 I got annoyed that running tests from one file took almost 13 minutes so I moved some of the tests out of the `TransactionTestCase` test classes that didn't need to be there, reducing the run time by 5 1/2 minutes on my machine

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/extras/tests/test_jobs.py [hunk 1]
```diff
@@ -35,9 +35,147 @@
 from nautobot.extras.models.models import JobLogEntry
 
 
-class JobTest(TransactionTestCase):
+class JobTest(TestCase):
     """
-    Test basic jobs to ensure importing works.
+    Test job features that don't require a transaction test case.
```
Comment (jathanism) [inline nautobot/extras/tests/test_jobs.py:40]: How can we reasonably determine which tests do or do not require transaction test cases?

Comment (gsnider2195) [reply nautobot/extras/tests/test_jobs.py:40]: I left anything that uses `create_job_result_and_run_job` in a transaction test case

**Comment by glennmatthews** [comment]
Latest CI runs don't look notably faster, but that may just fall within normal variance?

**Review by glennmatthews — APPROVED** [review]

**Comment by gsnider2195** [comment]
> Latest CI runs don't look notably faster, but that may just fall within normal variance?

I only tested it once locally but it was significantly faster. 780 seconds down to 460

**Pull request merged**